src/xb-database-manager.c

```diff
@@ -366,20 +439,23 @@ xb_database_manager_create_db_internal (XbDatabaseManager *self,
   XapianQueryParser *query_parser;
   DatabasePayload *payload;
   GFileMonitor *monitor;
+  const char *path;
 
-  g_assert (!g_hash_table_contains (priv->databases, xbdb.path));
+  path = xb_database_path (xbdb);
+
+  g_assert (!g_hash_table_contains (priv->databases, path));
+
+  if (xbdb.manifest_path)
```

Comment:
In one case, it's a helper function to get the "path", whatever that means, mostly for error messages. In the second, we actually care if it's a path or a manifest path, since we use the data from each differently.